src/base/bittorrent/sessionimpl.cpp

```diff
@@ -5033,38 +5032,39 @@ void SessionImpl::loadCategories()
 {
     m_categories.clear();
 
-    QFile confFile {(specialFolderLocation(SpecialFolder::Config) / CATEGORIES_FILE_NAME).data()};
-    if (!confFile.exists())
-    {
-        // TODO: Remove the following upgrade code in v4.5
-        // == BEGIN UPGRADE CODE ==
-        upgradeCategories();
-        m_needUpgradeDownloadPath = true;
-        // == END UPGRADE CODE ==
-
-//        return;
-    }
+    const int fileMaxSize = 1024 * 1024;
+    const Path path = specialFolderLocation(SpecialFolder::Config) / CATEGORIES_FILE_NAME;
 
-    if (!confFile.open(QFile::ReadOnly))
+    const auto readResult = Utils::Fs::readFile(path, fileMaxSize);
+    if (!readResult)
     {
-        LogMsg(tr("Failed to load Categories. File: \"%1\". Error: \"%2\"")
-               .arg(confFile.fileName(), confFile.errorString()), Log::CRITICAL);
+        if (readResult.error().status == Utils::Fs::ReadError::ErrorStatus::NotExist)
+        {
+            // TODO: Remove the following upgrade code in v4.5
+            // == BEGIN UPGRADE CODE ==
+            upgradeCategories();
+            m_needUpgradeDownloadPath = true;
+            // == END UPGRADE CODE ==
+            return;
+        }
+
+        LogMsg(tr("Fail to load Categories. %1").arg(readResult.error().message), Log::WARNING);
         return;
     }
 
     QJsonParseError jsonError;
-    const QJsonDocument jsonDoc = QJsonDocument::fromJson(confFile.readAll(), &jsonError);
+    const QJsonDocument jsonDoc = QJsonDocument::fromJson(readResult.value(), &jsonError);
     if (jsonError.error != QJsonParseError::NoError)
     {
-        LogMsg(tr("Failed to parse Categories configuration. File: \"%1\". Error: \"%2\"")
-               .arg(confFile.fileName(), jsonError.errorString()), Log::WARNING);
+        LogMsg(tr("Fail to parse Categories configuration. File: \"%1\". Error: \"%2\"")
```

Comment:
>Fail to....

Doesn't look right to me.

Reply:
> Doesn't look right to me.

Indeed.  It should be `Failed to` (past tense).

Reply:
OK, fixed.